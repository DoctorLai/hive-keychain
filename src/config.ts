const Config = {
  hiveEngine: {
    mainNet: 'ssc-mainnet-hive',
  },
  claims: {
    FREQUENCY: +(process.env.DEV_CLAIM_FREQUENCY || 10 * 60 * 1000),
    freeAccount: {
      MIN_RC_PCT: +(process.env.DEV_CLAIM_ACCOUNT_RC_PCT || 95),
      MIN_RC: +(process.env.DEV_CLAIM_ACCOUNT_MIN_RC || 1.1 * 10 * 10 ** 12), // 10% more than 10^13 (current creation cost)
    },
  },
};

export default Config;
