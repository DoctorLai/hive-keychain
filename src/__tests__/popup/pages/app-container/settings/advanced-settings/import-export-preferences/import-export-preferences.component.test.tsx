import { Icons } from '@popup/icons.enum';
import importExportPreferences from 'src/__tests__/popup/pages/app-container/settings/advanced-settings/import-export-preferences/mocks/import-export-preferences';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import config from 'src/__tests__/utils-for-testing/setups/config';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
config.byDefault();
describe('import-export-preferences.component tests:\n', () => {
  let _asFragment: () => DocumentFragment;
  const { methods, constants, extraMocks } = importExportPreferences;
  methods.afterEach;
  beforeEach(async () => {
    _asFragment = await importExportPreferences.beforeEach();
  });
  it('Must load import-export page and match snapshot', () => {
    expect(_asFragment()).toMatchSnapshot(constants.snapshotName.default);
  });
  it('Must open import window', async () => {
    await clickAwait([alButton.menuPreFix + Icons.IMPORT]);
    expect(extraMocks.spy.importSettings).toBeCalledTimes(1);
  });
  it('Must try to export settings file', async () => {
    extraMocks.getMultipleValueFromLocalStorage();
    extraMocks.aClick;
    await clickAwait([alButton.menuPreFix + Icons.EXPORT]);
    expect(extraMocks.spy.exportSettings).toBeCalledTimes(1);
    expect(extraMocks.aClick).toBeCalledTimes(1);
  });
});
