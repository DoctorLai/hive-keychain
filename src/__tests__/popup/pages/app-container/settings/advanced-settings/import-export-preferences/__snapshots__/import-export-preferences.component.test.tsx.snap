// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-export-preferences.component tests:
 Must load import-export page and match snapshot: import-export-preferences.component Default 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Import / Export Settings
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="import-export-preferences-page"
        >
          <div
            class="menu-page"
          >
            <div
              class="menu"
            >
              <div
                aria-label="menu-settings-button-file_upload"
                class="menu-item"
              >
                <span
                  class="icon-component material-icons-outlined icon "
                >
                  file_upload
                </span>
                <div
                  class="menu-label"
                >
                  Import Settings
                </div>
              </div>
              <div
                aria-label="menu-settings-button-file_download"
                class="menu-item"
              >
                <span
                  class="icon-component material-icons-outlined icon "
                >
                  file_download
                </span>
                <div
                  class="menu-label"
                >
                  Export settings
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
